parliament_challenge.py

```diff
@@ -0,0 +1,111 @@
+import requests
+import waitress
+import flask
+from flask import Flask, request, json
+
+
+def filter_speeches_dict(speeches):
+    """
+    Filter speeches data for relevant items.
+    """
+
+    speeches_list = []
+    for s in speeches:
+        speechesDict = {
+            "anforande_id": s.get("anforande_id"),
+            "dok_datum": s.get("dok_datum"),
+            "parti": s.get("parti"), 
+            "avsnittsrubrik": s.get("avsnittsrubrik"),
+            "links": [{ 
+                "rel": "speech",
+                "href": s.get("protokoll_url_www")
+                    }],
+            "intressent_id": s.get("intressent_id")
+            }
+        speeches_list.append(speechesDict)
+    return speeches_list
+
+
+def filter_member_dict(member):
+    """
+    Filter member data for relevant items.
+    """
+
+    membersDict = {
+        "tilltalsnamn": member.get("tilltalsnamn"),
+        "valkrets": member.get("valkrets"),
+        "bild_url_192": member.get("bild_url_192"),
+        "intressent_id": member.get("intressent_id")
+    }
+    return membersDict
+
+
+def get_member_data(intressent_id):
+    """
+    Get memeber data filter for relevant items.
+    """
+
+    domain = 'http://data.riksdagen.se'
+    members = 'personlista'
+    format_type = 'json'
+    response = requests.get('{}/{}/?iid={}&utformat={}'.format(domain, members, intressent_id, format_type))
+    if response.status_code == 200:
+        data = response.json()
+        member = data["personlista"]["person"]
+        filtered_member = filter_member_dict(member)
+        return filtered_member
+
+
+def get_speeches_data(anftyp, size):
+    """
+    Get speeches data and filter for relevant items.
+    Loop through filtered speeches to get "intressent id" 
+    and pass it as an argurment to the get_member_data 
+    to get member data.
```

Comment:
Simplify the explanation. Just say .. 

.... 
For each speech we query the member api to get the member data and then merge it with the speech data.
...

Reply:
You don't have to say which functions you're calling and what arguments you pass to it. The programmer or anyone reading your code will understand it from reading the code. The description should only be very high level .. in plain english what the function is supposed to do, what arguments it takes and what it returns.

Reply:
updated